Motion blur post process

ykafia opened the pull request:
# PR Details

<!--- Provide a general summary of your changes in the Title above -->
This PR adds 2 motion blur techniques as post processing effects

## Description

<!--- Describe your changes in detail -->

1. [Chapman](http://john-chapman-graphics.blogspot.com/2013/01/per-object-motion-blur.html)/[NVIDIA](https://developer.nvidia.com/gpugems/gpugems3/part-iv-image-effects/chapter-27-motion-blur-post-processing-effect)
2. [Guertin, McGuire et al](https://casual-effects.com/research/Guertin2014MotionBlur/index.html)

The first technique is a simple and straight forward technique, it does wonders without adding a lot of computation complexity but has many caveats for those who aims for a very photorealistic result namely :
* Assumes all motion is linear
* Static object in the front can bleed on moving objects in the back
* The silouhette of moving objects is maintained, the blur is only inside objects
* Since the computation of screen space velocity is pixel precise, it does not compute realistic blur

The second technique tries to solve some of the previous issues but still have some caveats. The good point is that it's currently used in Unreal Engine (albeit with some modifications of their own), I'd like to see how Stride would handle this technique.

## Related Issue

#1058
<!--- If suggesting a new feature or change, please discuss it in an issue first -->
<!--- If fixing a bug, there should be an issue describing it with steps to reproduce -->
<!--- Please link to the issue here: -->

## Motivation and Context

<!--- Why is this change required? What problem does it solve? -->
Motion blurs add to the realism of a scene and they look super cool!


## Types of changes

<!--- What types of changes does your code introduce? Put an `x` in all the boxes that apply: -->

- [ ] Docs change / refactoring / dependency upgrade
- [ ] Bug fix (non-breaking change which fixes an issue)
- [x] New feature (non-breaking change which adds functionality)
- [ ] Breaking change (fix or feature that would cause existing functionality to change)

## Checklist

<!--- Go over all the following points, and put an `x` in all the boxes that apply. -->
<!--- If you're unsure about any of these, don't hesitate to ask. We're here to help! -->

- [x] My change requires a change to the documentation.
- [ ] I have added tests to cover my changes.
- [ ] All new and existing tests passed.

Change in the documentation include adding a page describing the effect and the different parameters.

tebjan committed:
[Physics] Added setter to use a custom scene system to render the physics debug shapes

tebjan committed:
[Physics] Remove reference from PhysicsProcessor to SceneSystem to render physics debug shapes into any given scene

johang88 committed:
AssetMemberNode.ContentChanged only generates missing ids if needed

tebjan committed:
[Games] Allows to set a custom render target for the GameProfilingSystem

tebjan committed:
[Stride.Core.Mathematics] Containment method for Vector2 of RectangleF is now exactly the same as for individual x and y values. for some reason it was excluding the right and lower border.

tebjan committed:
[Graphics] SDL: Don't minimize fullscreen when focus is lost

tebjan committed:
[Games] Added SetSize to GameWindow to allow resizing the window by code

Eideren committed:
[SDL] Drop files and clipboard

Zeroto committed:
Fix to stop the visual studio installer exiting with code -1 during prerequisites install

The prerequisites installer somehow keeps the msiexec service in an installing state even though the installer is already finished and closed. This causes the visual studio installer to refuse to install because it thinks there is still an other installer running.

By installing the visual studio prerequisite before installing the other prequisites this problem is avoided. This is a hack because the windows installer state will still be wrong after this, but this will at least allow people to install stride properly without encountering an error that they can't easily solve. The windows installer being in a wrong state will most likely not cause any other problems for the user, because the next windows installer will detect the wrong state and fix it, or in the case of the visual studio installer it will fix it after hitting retry or restarting the visual studio installer.

The proper solution for not ending up in this state would be in the advanced installer script/version. However, I could not find any online solutions or other people with the same problem.

Eideren committed:
[Rendering] Instancing: fix instance back to scene (#871)

* [Rendering] Instancing: fix instance back to scene

* [Rendering] Instancing: fix crash when removing instance without render model

manio143 committed:
Allow using standard non-windows temporary directories

to enable project compilation with Stride NuGet packages under Linux

manio143 committed:
Remove WPF/Winforms reference for non-windows platforms

xen2 committed:
[Input] Put RawInput code inside #ifdef

xen2 committed:
Revert "[Input] Put RawInput code inside #ifdef"

This reverts commit 8e2424ab853d32ebdff5c39544dcfba410b0ca4f.

xen2 committed:
[Input] Put RawInput code inside #ifdef (fixed)

xen2 committed:
Merge pull request #888 from johang88/477_NormalMapTessellationFix

[#477] normalWS exposed when tessellation and normal mapping is used

xen2 committed:
Merge pull request #868 from Eideren/attenuation

[Shader] Expose point+spot attenuation and fix celshading

manio143 committed:
[Editor] Allow adding default configurations to GameSettings

manio143 committed:
[Editor] Improve the Add configuration label and positioning

manio143 committed:
[Editor] Cleanup GameSettings.Defaults view

manio143 committed:
[Editor] Fix command adding in GameSettingsNodeUpdater

manio143 committed:
[Editor] Make GameSettingsAsset.Defaults not ReadOnly to allow removal

manio143 committed:
Fix using uninitialized CharacterScript

This is because of the fix to Async script priority.

azeno committed:
[Input] Pass window (or control) to input sources explicitly to allow creating additional input sources for windows other than the main window

xen2 committed:
[Build] Switch to .NET 5.0 (fixes #912)

manio143 committed:
Add default RID to Windows project template

manio143 committed:
Initialize Windows project template with netcoreapp3.1 when using .NET Core

manio143 committed:
Add global.json to new projects

xen2 committed:
Merge pull request #910 from manio143/win_rid

Add default RID to Windows, TargetFramework and global.json project template

xen2 committed:
Merge pull request #907 from manio143/sample_game_space_escape_fix

Fix using uninitialized CharacterScript in SpaceEscape sample

xen2 committed:
Merge pull request #906 from manio143/gamesettings_add_configuration

[Editor] Allow adding default configurations to GameSettings

manio143 committed:
Fallback to net472 AssetCompiler first

Mobile projects don't target .NET Framework or .NET Core, but require MSBuild tools to be used rather than dotnet build tool

azeno committed:
[Graphics] When creating the swap chain use the handle as provided by WindowHandle

Assuming the window has a property called Handle which points to the HWND is not correct. This allows to use the Stride graphics API in other windowing frameworks.

Eideren committed:
[Threading] Temporary workaround for #913 (#916)

xen2 committed:
[Build] Switched MSBuild.Sdk.Extras to 2.1.6

xen2 committed:
[Input] Remove ambiguous reference between our IReadOnlySet<T> and .NET 5.0 one

azeno committed:
[Input] Input manager is no longer a game system. Allows it to be used in more scenarios.

azeno committed:
Adds missing ifdef to VS integration test so all tests compile without an error

xen2 committed:
Merge pull request #919 from vvvv/dev/azeno/multiple-window-input-sources

Support multiple window input sources

azeno committed:
[Input] Fixes compile error introduced in previous commit. Not sure why it wouldn't show up when I was running the tests locally.

xen2 committed:
[Build] Revert default editor framework to .NET FW for now

azeno committed:
[Build] Fixes Stride.Assimp.dll not found during package build by referencing the project directly

Seems like something changed in msbuild that the dll isn't copied anymore. I guess in the past it came through Stride.Importer.Assimp.vcxproj even though there it's marked as CopyLocal=false

azeno committed:
[Rendering] Allows to use in-memory generated materials as blend layers by simply keeping the material descriptor.

azeno committed:
[Rendering] Allows to read the in- and outputs of an image effect.

azeno committed:
[Engine] Use property key on graphics device to force present interval on all presenters while profiling

Will currently only work in Direct3D the implementation. Vulkan would require to re-create the swap chain while in OpenGL the present interval is controlled through the window? It's not even implementing the Present operation.

xen2 committed:
[Build] Explicitly list TFM in xunit.runner.stride, otherwise net461+net472 was causing issues during test build

AmbulantRex committed:
Check for XML documentation file and document dependencies

xen2 committed:
[Build] Rename StrideGraphicsApisDependentBuildAll into StrideGraphicsApiDependentBuildAll (fixes #940)

xen2 committed:
[Build] Fix net5.0-window to net5.0-windows7.0 mapping for Stride.Assets.Models

xen2 committed:
[Build] Bump MSBuild.Sdk.Extras to 3.0.22

azeno committed:
[Engine] Fixes memory leak in forward renderer. The cached depth stencil view needs to be disposed.

xen2 committed:
[Build] Reset a part of 3893ee271

xen2 committed:
[Build] Proper fix for package library path resolution (follow MSBuild.Sdk.Extras rules)

xen2 committed:
[Build] NuGet restore: improvements to error reporting and recovery in case Stride.ConnectionRouter.exe file is still running

azeno committed:
[Rendering] Fixes crash when enabling a color transform during runtime.

The previous code relied on UpdateEffect returning true when the set of enabled transforms changed, however that was not the case when the generated shader stayed the same.

azeno committed:
[Rendering] Fixes crash when changing the LightStreak.IsAnamorphic property during runtime

Basewq committed:
[Samples] Fix missing sub-scenes in CSharpBeginner that belongs to the Game project's root asset (fixes #821)

xen2 committed:
[Setup] Add .NET SDK 5.0.100+ to list of prerequisites

Basewq committed:
[Input] Fix detecting WinForms right shift key (fixes #754, fixes #929)

Basewq committed:
[Input] InputSystem should be public so external users can access it

azeno committed:
[Games] Adds option to use a borderless window as fullscreen mode (SDL only)

azeno committed:
[Games] Adds Alt+Enter (keypad) to fullscreen toggle shortcuts.

xen2 committed:
[Build] Update VC++ dependency check to VS2019

xen2 committed:
[Prerequisites] If Visual Studio Installer fails, display the list of missing workloads so that the user can install them himself.

azeno committed:
[Graphics] The back buffer format adjustment based on the device color space will also be applied on resize

- It will now also work for all pixel formats by using ToSrgb and ToNonSrgb extension methods
- Fixes format adjustment not being applied if color space was set to Gamma but device supported Linear space

xen2 committed:
Merge pull request #968 from vvvv/dev/azeno/sdl-window

[Games] A few minor window related additions (SDL)

xen2 committed:
Merge pull request #967 from vvvv/dev/azeno/physics-parent-scene

[Physics] Render physics debug shapes into any given scene

azeno committed:
Register assemblies in the assembly container to ensure later assembly loads succeed

azeno committed:
[Input] Unique input device ids per window as well as for simulated devices

xen2 committed:
[Build] Made .NET code cross-platform (no more #ifdef per platform)

tebjan committed:
[Graphics] Adds isSRgb parameter to CreateTextureFromNative. Allows to specify that a native texture is in gamma space.

tebjan committed:
[Graphics] Fixed isSRgb parameter for DX12

Eideren committed:
[Doc] More troubleshooting guidelines

Eideren committed:
[Doc] Small addition

Eideren committed:
[Doc] No vs build info

tebjan committed:
[VR] Updates OpenVR to 1.14.15

tebjan committed:
[Graphics] Refines texture initialization from native shader resource view. fixes #782 and adds to #459

xen2 committed:
Merge pull request #982 from vvvv/dev/tebjan/vr-initialization-fix

[VR] Initialization fix

Eideren committed:
[Threading] Pull latest semaphore perf improvements

Eideren committed:
[Threading] Support switching default threadpool

Eideren committed:
[Threading] Align naming scheme

xen2 committed:
Merge pull request #984 from Eideren/threading_tebjan

[Threading] Allow switching default pool + small performance improvement

xen2 committed:
[Setup] Switch from verisign to globalsign timestamp server

xen2 committed:
[Build] Remove beta04 tag

cstdbool committed:
Correct VS workload name and fix VS installer arguments

cstdbool committed:
Correct Stride Visual Studio extension GUID

xen2 committed:
[Build] Bump to 4.0.0.2

xen2 committed:
[Build] Adjust version info

xen2 committed:
[Build] Use revision number to store git height

xen2 committed:
[Build] Put back a simple -beta as suffix, and bump to 4.0.1

cstdbool committed:
Remove GameStudio dependencies when a build is uninstalled

xen2 committed:
[Samples] Bump package version

cstdbool committed:
Remove packages which are no longer referenced by any GameStudio package after uninstall or update

cstdbool committed:
[Rendering] InstancingRenderFeature: null check for VisibilityGroup

xen2 committed:
[Build] Always try to publish Stride.Samples.Templates (but ignore duplicates)

cstdbool committed:
[Editor] Update GameStudio Q&A link to the new "gamedev.stackexchange.com" link

cstdbool committed:
[Editor] Starting default Visual Studio IDE from .NET 5 GameStudio fails

cstdbool committed:
[Editor] Fix GameStudio crash when specifying the default IDE from settings

amerkoleci committed:
ENH: Update to latest Vortice.Vulkan (target 1.2.167 vulkan SDK).

xen2 committed:
[Build] Bump System.Runtime.CompilerServices.Unsafe to 5.0.0

manio143 committed:
Set TargetFramework to net5.0-windows for new projects

xen2 committed:
[Setup] Bumped to Advanced Installer 17.9 (due to Error message: 'Done Adding Additional Store', it seems to be fixed in 16.4+)

https://www.advancedinstaller.com/forums/viewtopic.php?f=2&t=31947&sid=14b7bc40b5b2cbcaddf66328cbb432ed&start=15#p118096

xen2 committed:
[Build] Update build script for Advanced Installer 17.9

xen2 committed:
[Build] AdvancedInstaller: use proper executable for command line build (otherwise it uses UI messagebox, blocking teamcity/CI)

xen2 committed:
[Build] Switched from globalsign to digicert timestamp server

xen2 committed:
[Graphics] Vulkan: fixed code since update to latest Vortice.Vulkan (#1005)

azeno committed:
[Rendering] Fixes null pointer after switching to an empty scene

xen2 committed:
[Build] 4.0: removed -beta tag

xen2 committed:
[Backers] Updated list of backers

xen2 committed:
[Build] Bump to 4.1!

xen2 committed:
[Build] Default tooling to NET 5.0

xen2 committed:
[Build] Removed NUnitLite references

xen2 committed:
[Build] Switch runtime to .NET 5.0 (instead of .NET FW and .NET Standard)

manio143 committed:
Update Stride.Assets.Presentation.csproj

xen2 committed:
[Build] Switch to "dotnet test" for running unit tests

xen2 committed:
[Build] Enable unit tests to work with teamcity and VS

xen2 committed:
[Tests] Fixed mscorlib to System.Private.CoreLib in unit tests

cstdbool committed:
[Editor] Improve TimeSpan property editor control

cstdbool committed:
[Editor] Add DateTime property editor control

cstdbool committed:
[Editor] Increase DateTimeEditor and TimeSpanEditor precision to OS tick level

xen2 committed:
[Build] Switched from Solution Dependencies to ProjectReference

xen2 committed:
[Build] Native: prefix folder with platform (for future unification of .NET 5 without RID)

xen2 committed:
Merge pull request #1014 from cstdbool/961_datetime_cannot_be_serialized

[Editor] Add DateTime property editor and improve TimeSpan property editor

cstdbool committed:
[Editor] Show property type on hover in property grid

xen2 committed:
[Build] Remove RuntimeIdentifier. Switched from MSBuild.Sdk.Extras to Microsoft.NET.Sdk since we don't need RuntimeIdentifier anymore (backported StrideGraphicsApi targets to work on top of Microsoft.NET.Sdk)

xen2 committed:
[Assets] CompilerApp: Consider both net5.0-windows and net5.0-windows7.0 for asset compiler path

xen2 committed:
[GameStudio] Tests: fix build

xen2 committed:
[Build] Bump C# to 9.0

cstdbool committed:
[Editor] Resizing property grid - property name and type should be responsive

xen2 committed:
Merge pull request #1015 from cstdbool/963_show_inspector_property_datatype

[Editor] Show property type on hover in property grid

xen2 committed:
[Build] Package & ssdeps: use BuildOutputInPackage instead of TfmSpecificPackageFile so that we don't have to guess the final target path generated in PackTask (which can sometimes be differnt than expected, i.e. net5.0-windows7.0 rather than net5.0-windows)

Basewq committed:
[Core] Fix Android crash due to GetApplicationBinaryDirectory call where path executable path was null on Android, and instead use the path of the installed apk.

xen2 committed:
[Build] Reuse some part of MSBuild.Sdk.Extras to reenable support for UWP, Android and iOS

xen2 committed:
[Tests] Graphics: use RenderDoc only on .NET platforms

xen2 committed:
[Build] Moved Stride.Debugger to Editor solution folder

xen2 committed:
[Assets] Disabled a failing unit test

cstdbool committed:
[Launcher] Distinguish between dev updates and official updates

cstdbool committed:
[Editor] Hide buttons that modify archetype reference when it is set to read-only

xen2 committed:
[Build] Switched tools to single TFM (managed C++ projects were broken with 16.9)

xen2 committed:
[Build] Force .NET version so that NuGet doesn't get upgraded behind our back. Later, we should probably use a AssemblyLoadContext to load either NuGet or MSBuild so that they can have different versions.

xen2 committed:
[Build] Bump MSBuildLocator version

xen2 committed:
[Build] Make sure all files are packaged (list of ignored extensions)

ykafia committed:
Merge pull request #1 from stride3d/master

[Build] Force .NET version so that NuGet doesn't get upgraded behind …

xen2 committed:
Merge branch 'feature/net5_only'

ykafia committed:
Merge pull request #2 from stride3d/master

update master

aunpyz committed:
UIComponent can set sampler

D3ZAX committed:
Add ACES (#1037)

Add ACES tone mapping operator

ykafia committed:
Merge pull request #4 from stride3d/master

Add ACES (#1037)

herocrab committed:
Added outline post fx.

herocrab committed:
Added fog postfx.

xen2 committed:
Updated README.md for joining .NET Foundation

xen2 committed:
Updated LICENSE.md for .NET Foundation

herocrab committed:
Updates for PR, contributors and curly braces.

herocrab committed:
Updated for PR, stride contributors and curly braces.

herocrab committed:
Removed reference to Silicon Studio.

aunpyz committed:
Apply sampler state on 'Clip To Bounds' checked

aunpyz committed:
Add Anisotropic option to match SpriteSampler option

xen2 committed:
Merge pull request #1045 from aunpyz/feature/ui-sampling

Add sampling option to UIComponent

xen2 committed:
Merge pull request #1039 from herocrab/add_fog_postfx

Add Fog image effect to post processing effects system

xen2 committed:
Merge branch 'master' into add_outline_postfx

xen2 committed:
Merge pull request #1038 from herocrab/add_outline_postfx

Add Outline image effect to post processing effects system

xen2 committed:
[Rendering] Fixed some file headers

xen2 committed:
Revert "Add sampling option to UIComponent"

xen2 committed:
[Build] Added global.json to Stride.Core.Tasks project (#1051)

xen2 committed:
Updated copyright headers

xen2 committed:
Merge pull request #1042 from xen2/dotnet_fdn

Joining .NET Foundation

ykafia committed:
Motion Blur initial commit. TileMax

ykafia committed:
Merge branch 'master' of https://github.com/ykafia/stride

manio143 committed:
Improve usability of AppSettings

manio143 committed:
Fix Yaml deserialization and use IEnumerable approach

manio143 committed:
Comments and cleanup

ykafia committed:
Champan progress

ykafia commented:
Just for information the first motion blur technique is finished, it works, parameters works. I want to finish implementing the second one but i could move it into another PR ?

manio143 commented on sources/engine/Stride.Rendering/Rendering/Images/MotionBlur/IMotionBlur.cs:
```diff
@@ -0,0 +1,11 @@
+{
+    public interface IMotionBlur : IImageEffect
+    {
+        bool NeedRangeDecompress { get; }
```
Seems you're not referencing this property anywhere - also naming wise, should it follow the `Requires*` naming convention or is `Need` something else?

ykafia replied:
I've copy pasted the fields from the anti aliasing interface but you're right this property is not used, i removed it in my last commit!

ykafia committed:
Correction Chapman blur

ykafia commented:
Just something i noticed, and maybe due to my lack of math knowledge but the MeshVelocity computes velocity of animated meshes (skinning) but not when they are translated/rotated. Shader wise, the math seems to check out, i'm not sure if it's the intended behavior but i'll try to see if something can be different on the TransformRenderFeature

xen2 committed:
Merge pull request #1055 from manio143/appSettingsUsability

[Core] Improve AppSettings usability

xen2 committed:
[Assets] MSBuild is now loaded with an AssemblyLoadContext, so that there is no more NuGet version conflicts. As a result, specific .NET version (with global.json) is not required anymore.

xen2 committed:
[Install] No need for custom Visual Studio packages anymore, .NET SDK is enough.

xen2 committed:
[Build] Avoid building twice platforms with only one graphics API (MSBuild race conditions)

xen2 committed:
[Launcher] Removed hardcoded list of TFM (doesn't work if new package doesn't have .NET 4.7 anymore; also we can later easily bump TFM to .NET 6 without having to change the launcher)

xen2 committed:
[Launcher] Don't use PublishTrimmed (https://github.com/microsoft/microsoft-ui-xaml/issues/3749)

xen2 committed:
[Launcher] Publish: Also use IncludeAllContentForSelfExtract

ykafia committed:
Updates on a working blur. Need better computation + slow loading

xen2 committed:
[Assets] Package upgrader can migrate project from .NET Framework to .NET 5.0

xen2 committed:
[Samples] Bumped samples

ykafia committed:
working blur implementation, needs  rework

ykafia committed:
Update on better algorithm, needs better tile blurring

ykafia committed:
Documentation update

ykafia committed:
Merge branch 'stride3d:master' into master

ykafia committed:
Merge branch 'master' of https://github.com/ykafia/stride

ykafia committed:
update branch

merge

ykafia committed:
Merge branch 'master' into motionBlurPostProcess

xen2 committed:
[Launcher] NuGet: specify a default TargetFramework if the package doesn't have any (fixes #1067)

ykafia committed:
Correction chapman motion blur values

xen2 committed:
[Setup] Bumped version to 5.0.5

manio143 committed:
Prevent ArgException of Path.Combine in Single File Publish scenario

johang88 committed:
#1063 GameSettingsAssetCompiler dynamically detecs runtime content types (#1064)

aunpyz committed:
Make enum name more descriptive & add missing SamplerState case (#1050)

azeno committed:
[Reflection] Give the deps.json file a higher priority when loading an assembly by name

The deps.json file preserves the compilation context - if it exists we can be sure that the assembly it points by simple name is the correct one. Otherwise we might end up loading multiple versions of the same assembly.

xen2 committed:
[Targets] MSBuild MakeRelative doesn't include trailing \ anymore (https://github.com/dotnet/msbuild/issues/6493). Thanks manio143 for the fix!

azeno committed:
[Reflection] Loads mixed-mode assemblies containing fixups from file instead of from byte array (fixes issue #1066)

joreg committed:
[Input] fixes PointerPoint.Id always being 0

tebjan committed:
[Graphics] Fixes restriction of fromData of SetData if Texture. The size of the data can be bigger than the target, it doesnt need to match eaxactly.

AmbulantRex committed:
Merge branch 'master' into AmbulantRex/user-xml-documentation

AmbulantRex committed:
Pre-caching documentation on start of session.

AmbulantRex committed:
Pre-caching documentation on creation of new session

xen2 committed:
[Build] Fix build with .NET SDK 5.0.300+

Details: https://docs.microsoft.com/en-us/dotnet/core/compatibility/sdk/5.0/referencing-executable-generates-error

xen2 committed:
[Assets] Improve package upgrader to better handle the case of .NET framework libraries

azeno committed:
[Graphics] Disposes the constant buffers allocated by an effect instance

Basewq committed:
[Engine] Fix InstancingProcessor dependency on ModelRenderProcessor where ModelRenderProcessor might not be created yet (fixes #813) (#991)

Co-authored-by: Basewq <Basewq@users.noreply.github.com>

Commit:
Improve the editor gizmos

Commit:
small fix

Commit:
added the option to use linear movement for rotating

xen2 committed:
Merge pull request #1083 from salahchafai/GizmoImprovements

Improve the editor gizmos

Commit:
improve the viewport grid

Eideren committed:
[Math] Added a Goto method to Vectors (cont. from @VexedUXR in #1089) (#1090)

* Added a "Goto" method to vectors

* Updated rest of the API to support pass by 'in'.

Co-authored-by: VexedUXR <randomstuff961@gmail.com>

xen2 committed:
[Launcher] packages.stride3d.net is deprecated (only for Xenko 3.0 and lower) so install it only if explicitly requested by user

Eideren committed:
[Physics] Improve collision and contact harvesting

Eideren committed:
[Physics] Add colliding check to collision pair

Eideren committed:
[Math] Math -> MathF (#1099)

xen2 committed:
Merge pull request #1100 from Eideren/physics_perf

[Physics] Improve collision and contact harvesting

manio143 committed:
[Editor] Don't skip first level directories for code assets (#848)

Eideren committed:
[Physics] Explicitly disable CCD when flag is not set

Eideren committed:
[Physics] Fix collision harvesting with complex types

Eideren committed:
[Physics] Push collisions into respective collider's collision store

Eideren committed:
[Physics] Deprecate 'ProcessCollisions', physics change rendered this performance improvement obsolete

Eideren committed:
[Graphics] Log missing shader generic argument error

Eideren committed:
[Graphics] Fix missing generics when setting material shader classes

Eideren committed:
[Physics] Collision: struct doesn't generate equality operator by default, add it

Eideren committed:
[Physics] Ignore static-static collision by default

Eideren committed:
[Physics] Documentation for new Simulation properties

Eideren committed:
[Physics] Ctor for Collision public

Eideren committed:
[Physics] Raycast and ShapeSweep can filter out triggers, true by default

xen2 committed:
Merge pull request #1104 from Eideren/graphics_gen_fix

[Graphics] Fix shader generic support

xen2 committed:
Merge pull request #1103 from Eideren/physics_fix

[Physics] Couple of minor collision related fixes following #1100

Eideren committed:
[Physics] Improve clarity of 'HitNoContResp'

Eideren committed:
[Physics] Refactor internal intersection callbacks

Eideren committed:
[Physics] Improve intersection test summaries

AmbulantRex committed:
Introduced concurrency measures to UserDocumentationService

AmbulantRex committed:
Flattened logic nesting in UserDocumentationService

Eideren committed:
[DOCS] Readme: remove .NET5-specific instructions

.NET5 is by default right now.

AmbulantRex committed:
Fix for #1102

AmbulantRex committed:
Correcting doc cache population mistake

AmbulantRex committed:
Converted empty array initialization to Array.Empty<T>. Addresses #608

AmbulantRex committed:
Merge branch 'master' into AmbulantRex/user-xml-documentation

xen2 committed:
Merge pull request #1105 from Eideren/raycast_triggers

[Physics] Raycast and ShapeSweep can filter out triggers, true by default

AmbulantRex committed:
Update UserDocumentationService.cs

xen2 committed:
Merge pull request #933 from AmbulantRex/AmbulantRex/user-xml-documentation

Check for XML documentation file and document dependencies. Fixes #917

AmbulantRex committed:
Changed NuGet restore to use friendly names

ykafia committed:
Merge pull request #9 from stride3d/master

Update

ykafia committed:
Merge branch 'motionBlurPostProcess' of https://github.com/ykafia/stride into motionBlurPostProcess

ykafia committed:
Little update
